modules/gapi/test/s11n/gapi_s11n_tests.cpp

```diff
@@ -45,6 +46,49 @@ template<> struct CompileArgTag<MyCustomType> {
 } // namespace detail
 } // namespace cv
 
+namespace {
+    class MyRMatAdapter : public cv::RMat::Adapter {
+    cv::Mat m_mat;
+    int m_value;
+    std::string m_str;
+public:
+    MyRMatAdapter(cv::Mat m, int value, const std::string& str)
+        : m_mat(m), m_value(value), m_str(str)
+    {}
+    virtual cv::RMat::View access(cv::RMat::Access access) override {
+        if (access == cv::RMat::Access::W) {
+            return cv::RMat::View(cv::descr_of(m_mat), m_mat.data, m_mat.step);
+        } else {
+            return cv::RMat::View(cv::descr_of(m_mat), m_mat.data, m_mat.step);
+        }
+    }
+    virtual cv::GMatDesc desc() const override { return cv::descr_of(m_mat); }
+    virtual void serialize(cv::gapi::s11n::IOStream& os) override {
+        os << m_value << m_str;
+    }
+    int getVal() { return m_value; }
+    std::string getStr() { return m_str; }
+};
+}
+
+namespace cv {
+namespace gapi {
+namespace s11n {
+namespace detail {
+    template<> struct S11N<MyRMatAdapter> {
+        static MyRMatAdapter deserialize(IIStream &is) {
+            int val;
+            std::string str;
+            is >> val >> str;
+            MyRMatAdapter p(cv::Mat(), val, str);
+            return p;
+        }
+    };
+} // namespace detail
+} // namespace s11n
+} // namespace gapi
+} // namespace cv
```

Comment:
I'd replace it with `deserialize` implemented right in the adapter code.

Reply:
Replaced